File: patterns-java/src/test/java/me/freedom4live/patterns/java/factory/method/CrazyCatFactoryTest.java
package me.freedom4live.patterns.java.factory.method;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class CrazyCatFactoryTest {

    private AbstractCatFactory catFactory;


    @BeforeEach
    void setUp() {
        catFactory = new CrazyCatFactory();
    }

    @Test
    void createCat_crazyCat_hasDifferentMeows() {
        //arrange
        Cat cat = catFactory.createCat();
        String previous = cat.meow();

        for (int i = 0; i < 1000; i++) {
            //act
            String current = cat.meow();

            //assert
            Assertions.assertNotEquals(previous, current);
            previous = current;
        }
    }
}

File: patterns-kotlin/src/test/kotlin/me/freedom4live/patterns/kotlin/factory/method/CrazyCatFactoryTest.kt
package me.freedom4live.patterns.kotlin.factory.method

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class CrazyCatFactoryTest {

    private lateinit var catFactory: AbstractCatFactory


    @BeforeEach
    fun setUp() {
        catFactory = CrazyCatFactory()
    }

    @Test
    fun createCat_createsCat_hasDifferentMeows() {
        //arrange
        val cat = catFactory.createCat()

        //act
        var previous = cat.meow()

        repeat(1000) {
            val current = cat.meow()

            //assert
            Assertions.assertNotEquals(previous, current)

            previous = current
        }
    }
}
